timeboost-sequencer/src/decrypt.rs

```diff
@@ -429,9 +433,15 @@ impl Worker {
             return false;
         };
         !share.is_empty()
-            && share
-                .iter()
-                .all(|shares| shares.len() >= self.keyset.one_honest_threshold().get())
+            && share.iter().all(|shares| {
+                let num_valid_shares = shares.iter().filter(|s| s.is_some()).count();
+                let num_invalid_shares = shares.len() - num_valid_shares;
+
+                // for valid decryption shares, as long as reaching f+1, we may try to hatch
+                // for invalid ones, we need 2f+1 to ensure consensus (to ignore invalid ciphertext)
+                num_valid_shares >= self.keyset.one_honest_threshold().get()
+                    || num_invalid_shares >= self.keyset.honest_majority_threshold().get()
+            })
```

Comment:
@twittner pls check my understanding is correct here

Reply:
Not sure. One thing that puzzles me is that there is no check that a peer has sent a decryption share batch for a round already, which means that a single node could add as many (most likely invalid shares) as it wants. Does this need to be protected against?

Reply:
good point, this is indeed problematic, i will fix it.

Reply:
fixed in [ca95972](https://github.com/EspressoSystems/timeboost/pull/346/commits/ca9597220493eea6d055eadde87e3f3a733e098e)